### PR TITLE
Add blog post on running FerretDB and CloudNativePG on Kubernetes

### PR DESCRIPTION
**Pull request by Fashander**
# Description

Closes FerretDB/engineering#177

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [ ] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [ ] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

**Review by chilagrow — COMMENTED** [review]
Please check the namespace consistency across the blog and test commands :pray:

File: website/blog/2024-06-20-run-ferretdb-cloudnativepg-kubernetes.md [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+#!/bin/sh
+
+# Generate a random password
+PASSWORD="$(dd if=/dev/urandom bs=1 count=12 status=none | base64 | tr -dc 'a-zA-Z0-9' | head -c 12)"
+
+kubectl create secret generic ferretuser-secret \
+ --from-literal=username=ferretuser \
+ --from-literal=password="${PASSWORD}" \
+ --type=kubernetes.io/basic-auth \
+ --namespace=cnpg
```
[the remark this answers is not in the document]
Comment (Fashander) [reply website/blog/2024-06-20-run-ferretdb-cloudnativepg-kubernetes.md:103]: Thanks @chilagrow. I've updated it— the initial namespace should be `cnpg` not `cnpg-system`

File: website/blog/2024-06-20-run-ferretdb-cloudnativepg-kubernetes.md [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+Create a temporary `mongosh` client to connect to the FerretDB instance.
+
+```sh
+kubectl -n cnpg run mongosh-client --image=rtsp/mongosh --rm -it -- bash
+```
+
+That will start up a `mongosh-client` within the namespace.
+
+```sh
+mongosh 'mongodb://ferretuser:<password>@ferretdb-service:27017/ferretdb?authMechanism=PLAIN'
```
Comment (chilagrow) [inline website/blog/2024-06-20-run-ferretdb-cloudnativepg-kubernetes.md:258]: This wouldn't work we didn't create `ferretdb-service`, could it be `ferretdb`. Also above output from `kubectl get cluster,pod,svc,secret -n cnpg` seems outdate.

File: website/blog/2024-06-20-run-ferretdb-cloudnativepg-kubernetes.md [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+   2024-06-20T01:40:21.558Z: Please star us on GitHub: https://github.com/FerretDB/FerretDB.
+   2024-06-20T01:40:21.558Z: The telemetry state is undecided.
+   2024-06-20T01:40:21.558Z: Read more about FerretDB telemetry and how to opt out at https://beacon.ferretdb.com.
+------
+ferretdb>
+```
+
+### Test CRUD commands in `mongosh`
+
+```json5
```
Comment (chilagrow) [inline website/blog/2024-06-20-run-ferretdb-cloudnativepg-kubernetes.md:288]: Let's follow the guide https://github.com/FerretDB/FerretDB/blob/main/website/docs/contributing/writing-guide.md#code-blocks

File: website/blog/2024-06-20-run-ferretdb-cloudnativepg-kubernetes.md [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+1
+```
+
+### Connect to Postgres cluster via `psql`
+
+Want to see what the data looks like in Postgres?
+Let's create a temporary `postgres-client` pod to connect to the `postgres-cluster-1` pod.
+
+```sh
+kubectl exec -it postgres-cluster-1 -n cnpg -- /bin/bash
```
Comment (chilagrow) [inline website/blog/2024-06-20-run-ferretdb-cloudnativepg-kubernetes.md:322]:
```
ferretdb> kubectl exec -it postgres-cluster-1 -n cnpg -- /bin/bash
Uncaught:
SyntaxError: Missing semicolon. (1:7)

> 1 | kubectl exec -it postgres-cluster-1 -n cnpg -- /bin/bash
    |        ^
  2 |

```

**Review by chilagrow — COMMENTED** [review]
@Fashander I think you've renamed something and few commands are not working, could you go over the entire doc and ensure commands work please :hugs:

File: website/blog/2024-06-20-run-ferretdb-cloudnativepg-kubernetes.md (hunk 3 above)
Comment (chilagrow) [reply website/blog/2024-06-20-run-ferretdb-cloudnativepg-kubernetes.md]: @Fashander the guide indicates request should be `js` block and response should be `json5`. Why not follow what it says?

**Review by chilagrow — COMMENTED** [review]
Let's follow the writing guide..

File: website/blog/2024-06-20-run-ferretdb-cloudnativepg-kubernetes.md (hunk 3 above)
Comment (chilagrow) [reply website/blog/2024-06-20-run-ferretdb-cloudnativepg-kubernetes.md]: The reason for following the guide is the request should be reformatted to make sure the consistency. The response should be as is from the mongosh.

File: website/blog/2024-06-24-run-ferretdb-cloudnativepg-kubernetes.md [hunk 5]
```diff
@@ -0,0 +1,410 @@
...
+
+## Prerequisites
+
+Ensure you have the following tools installed:
+
+- Kubernetes Cluster (You can set one up locally with [minikube](https://minikube.sigs.k8s.io/))
+- [`kubectl`](https://kubernetes.io/docs/reference/kubectl/)
+- [Helm](https://helm.sh/)
+
+Run the following command to install Helm:
```
Comment (rumyantseva) [inline website/blog/2024-06-24-run-ferretdb-cloudnativepg-kubernetes.md:33]: Two lines above you say I must have Helm installed :)
So, do I need helm installed, or will I install it as a part of this guide? :)

File: website/blog/2024-06-24-run-ferretdb-cloudnativepg-kubernetes.md [hunk 6]
```diff
@@ -0,0 +1,410 @@
...
+        app: ferretdb
+    spec:
+      containers:
+        - name: ferretdb
+          image: ghcr.io/ferretdb/ferretdb:latest
+          ports:
+            - containerPort: 27017
+          env:
+            - name: FERRETDB_POSTGRESQL_URL
+              value: 'postgresql://ferretuser@postgres-cluster-rw.cnpg.svc.cluster.local:5432/ferretdb'
```
Comment (rumyantseva) [inline website/blog/2024-06-24-run-ferretdb-cloudnativepg-kubernetes.md:187]: Why don't we need password in this connection string?

Comment (chilagrow) [reply website/blog/2024-06-24-run-ferretdb-cloudnativepg-kubernetes.md:185]: Because unauthenticated access is not desired.

Comment (rumyantseva) [reply website/blog/2024-06-24-run-ferretdb-cloudnativepg-kubernetes.md:185]: Ah, got it! we use `PLAIN` authentication in this example, maybe it is worth mentioning it for clarity. 
@chilagrow thanks!

File: website/blog/2024-06-24-run-ferretdb-cloudnativepg-kubernetes.md [hunk 7]
```diff
@@ -0,0 +1,410 @@
...
+(2 rows)
+ferretdb=> select * from testcollection_c141f891;
+                                                                                                                        _jsonb
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+ {"$s": {"p": {"_id": {"t": "objectId"}, "age": {"t": "int"}, "city": {"t": "string"}, "name": {"t": "string"}}, "$k": ["_id", "name", "age", "city"]}, "_id": "66741445f0f21e98f96badea", "age": 35, "city": "Chocolate Factory", "name": "Charlie"}
+(1 row)
+```
+
+## Clean up resources
+
```
Comment (rumyantseva) [inline website/blog/2024-06-24-run-ferretdb-cloudnativepg-kubernetes.md:393]: Maybe also a quick explanation message might be added. Something like, "when we don't need FerretDB nor Postgres cluster anymore, we can just cleanup the whole environment completely". 

Btw, does `minikube delete` delets the whole cluster? do we need to delete a particular namespace if we are going to delete the whole cluster?

File: website/blog/2024-06-24-run-ferretdb-cloudnativepg-kubernetes.md (hunk 5 above)
Comment (chilagrow) [reply website/blog/2024-06-24-run-ferretdb-cloudnativepg-kubernetes.md]: We how how to install `helm` but not `kubectl` and others. What is special about `helm`?

**Review by chilagrow — COMMENTED** [review]
See comment

**Review by chilagrow — DISMISSED** [review]
Nice post!

**Review by AlekSi — DISMISSED** [review]
LGTM, let's bump the date.

**Comment by Fashander** [comment]
@AlekSi PTAL

**Review by AlekSi — APPROVED** [review]

**Pull request merged**